File: Lesson - 10/1.cpp
#include <cmath>

int factors(int number) {
	int result = 0;
	long i = 1;
	while (i * i < number) {
	    if (number % i == 0) {
	        result += 2;
	    }
	    i++;
	}
	if (i * i == number)
	    result++;
	return result;
}

int solution(int N) {
    return factors(N);
}

File: Lesson - 11/1.js
function solution(A) {
    let result = [];
    let aSize = A.length;
    let occurenceMapping = new Map();
    
    for (const a of A) {
        if (!occurenceMapping.has(a)) {
            occurenceMapping.set(a, 1);
        }
        else {
            occurenceMapping.set(a, occurenceMapping.get(a) + 1);
        }
    }
    
    for (let i = 0; i < aSize; i++) {
        let totalDivisors = 0;
        const root = Math.sqrt(A[i]);

        for (let divi = 1; divi <= root; ++divi) {
            if (A[i] % divi == 0) {
                const div = A[i] / divi;
                
                if (occurenceMapping.has(divi)) {
                    totalDivisors += occurenceMapping.get(divi);
                }
                if (divi != div && occurenceMapping.has(div)) {
                    totalDivisors += occurenceMapping.get(div);
                }
            }
        }
        result.push(aSize - totalDivisors);
    }
    return result;
}